### PR TITLE
CHAOSPLT-531: Add a config for the minimum cron frequency

### PR DESCRIPTION
**Pull request by ptnapoleon**
## What does this PR do?

- [ ] Adds new functionality
- [x] Alters existing functionality
- [ ] Fixes a bug
- [ ] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- One might consider a disruption that runs every minute to be too often. One might even want to prevent that. Now we can!

## Code Quality Checklist

- [ ] The documentation is up to date.
- [ ] My code is sufficiently commented and passes continuous integration checks.
- [ ] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [ ] I manually tested the following steps:
    - `x`
    - [ ] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

File: api/v1beta1/disruption_cron_webhook.go [hunk 1]
```diff
@@ -196,3 +203,36 @@ func (d *DisruptionCron) validateDisruptionCronSpec() error {
 
 	return nil
 }
+
+func (d *DisruptionCron) validateMinimumFrequency(minFrequency time.Duration) error {
```
Comment (aymericDD) [inline api/v1beta1/disruption_cron_webhook.go:207]: The Schedule is part of the disruption cron spec. Is it better to do it in the `validateDisruptionCronSpec` method instead? 
The `validateDisruptionCronSpec`  already checks if the `spec.Schedule` is valid.  
WDYT of calling the `validateMinimumFrequency` in the `validateDisruptionCronSpec` method instead?

File: api/v1beta1/disruption_cron_webhook.go [hunk 2]
```diff
@@ -196,3 +203,36 @@ func (d *DisruptionCron) validateDisruptionCronSpec() error {
...
+	schedule, err := cron.ParseStandard(d.Spec.Schedule)
+	if err != nil {
+		return fmt.Errorf("spec.Schedule must follow the standard cron syntax: %w", err)
+	}
+
+	specDuration := defaultDuration
+	if d.Spec.DisruptionTemplate.Duration.Duration() > 0 {
```
Comment (aymericDD) [inline api/v1beta1/disruption_cron_webhook.go:214]: To cover this case, can you add this test in the `success cases` of the `ValidateCreate` method:

``` golang
When("the disruption template duration is greater than 0", func() {
    It("should validate the disruption cron successfully", func() {
	    // Arrange
	    disruptionCron := makeValidDisruptionCron()
	    disruptionCron.Spec.DisruptionTemplate.Duration = "1s"
    
	    disruptionCronJSON, err := json.Marshal(disruptionCron)
	    Expect(err).ShouldNot(HaveOccurred())
    
	    expectedAnnotation := map[string]string{
		    EventDisruptionCronAnnotation: string(disruptionCronJSON),
	    }
    
	    By("sending the EventDisruptionCronCreated event to the broadcast")
	    mockEventRecorder := mocks.NewEventRecorderMock(GinkgoT())
	    mockEventRecorder.EXPECT().
		    AnnotatedEventf(
			    disruptionCron,
			    expectedAnnotation,
			    Events[EventDisruptionCronCreated].Type,
			    string(EventDisruptionCronCreated),
			    Events[EventDisruptionCronCreated].OnDisruptionTemplateMessage,
		    )
    
	    disruptionCronWebhookRecorder = mockEventRecorder
    
	    // Act
	    warnings, err := disruptionCron.ValidateCreate()
    
	    // Assert
	    Expect(warnings).To(BeNil())
	    Expect(err).ShouldNot(HaveOccurred())
    })
})
```

**Review by aymericDD — APPROVED** [review]
💯 

I let some comments with suggestions, but nothing required.  🙇

**Comment by ptnapoleon** [comment]
This needs to check in ValidateUpdate, if the user just altered the frequency to be more/too often

File: api/v1beta1/disruption_cron_webhook_test.go [hunk 3]
```diff
@@ -498,6 +605,24 @@ var _ = Describe("DisruptionCron Webhook", func() {
 				})
 			})
 
+			When("the schedule frequency becomes shorter", func() {
+				It("should not return an error", func() {
```
Comment (aymericDD) [inline api/v1beta1/disruption_cron_webhook_test.go:609]:
```suggestion
				It("should return an error", func() {
```

**Review by aymericDD — APPROVED** [review]
💯

File: api/v1beta1/disruption_cron_webhook.go (hunk 1 above)
Comment (aymericDD) [reply api/v1beta1/disruption_cron_webhook.go:216]: Thanks for the explanations 🙇 
It makes sense for me 👍

**Pull request merged**